server/datastore/mysql/policies.go

```diff
@@ -760,20 +760,34 @@ func cleanupPolicyMembershipOnPolicyUpdate(ctx context.Context, db sqlx.ExecerCo
 // cleanupPolicyMembership is similar to cleanupPolicyMembershipOnPolicyUpdate but without the platform constraints.
 // Used when we want to remove all policy membership.
 func cleanupPolicyMembership(ctx context.Context, db sqlx.ExecerContext, policyID uint) error {
+	// delete all policy membership for the policy
 	delStmt := `
-	DELETE
-	  pm
-	FROM
-	  policy_membership pm
-	LEFT JOIN
-	  hosts h
-	ON
-	  pm.host_id = h.id
-	WHERE
-	  pm.policy_id = ?`
+		DELETE
+			pm
+		FROM
+			policy_membership pm
+		LEFT JOIN
+			hosts h
+		ON
+			pm.host_id = h.id
+		WHERE
+			pm.policy_id = ?
+	`
 
 	_, err := db.ExecContext(ctx, delStmt, policyID)
-	return ctxerr.Wrap(ctx, err, "cleanup policy membership")
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "cleanup policy membership")
+	}
+
+	// delete all policy stats for the policy
+	delCacheStmt := `DELETE FROM policy_stats WHERE policy_id = ?`
```

Comment:
good call, the rows could be locked while the cron is running.  Do you think a retry here would be sufficient?

Reply:
I wrapped the delete in a retry which has the risk of adding latency to the PATCH request if the
cron is updating those rows, but I think a better experience because it's more likely the delete
will occur after the cron updates.